src/pages/index.js

```diff
@@ -119,45 +181,129 @@ function closeModalEscape(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitButton = evt.submitter
+  submitButton.textContent = "Saving...."
+
+  api
+  .editUserInfo({ name: editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+  .then((userData) => {
+
+    profileName.textContent = userData.name;
+    profileDescription.textContent = userData.about;
+    closeModal(editModal);
+  })
+  .catch(console.error)
+  .finally(() => {
+   submitButton.textContent = "Save";''
+
+  });
+
 }
 
-function handleAddCardSubmit(evt) {
+function handleSubmit(evt) {
   evt.preventDefault();
   const inputValues = {
     name: cardNameInput.value,
     link: cardLinkInput.value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(addCardModal);
+  const submitButton = evt.submitter;
+  SetButtonText(true, submitButton);
+
+  api
+  .addCard(inputValues)
+  .then((newCard) => {
+    const newCardElement = getCardElement(inputValues);
+    cardsList.prepend(newCardElement);
+    evt.target.reset();
+    disableButton(cardSubmitButton, validationConfig);
+    closeModal(addCardModal);
+
+  })
+  .finally(() => {
+    SetButtonText(false, submitButton);
+
+  });
+
+
+}
+
+
+function handleAvatarSubmit(evt)  {
+  evt.preventDefault();
+    api.editAvatarInfo(avatarInput.value).then((avatarData) => {
+      profileImage.src = avatarData.avatar;
+      disableButton(avatarSubmitButton, validationConfig);
+      closeModal(avatarModal);
+
+    });
+
+
+}
+
+function handleDeleteCardSubmit(evt) {
+  evt.preventDefault();
+    api.deleteCard(selectedCardId).then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+
+    });
+
+}
+
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleLike(evt, id) {
+  const cardLikeBtn = evt.target;
+  const isLiked = cardLikeBtn.classList.contains("card__like-btn_liked");
+
+  api
+    .changeLikeStatus(id, isLiked)
+    .then((isLiked) => {
+      if (isLiked) {
+        isLiked = true;
```

Comment:
` isLiked = true;` is not needed becasue you don't need to reassign it - it has no effect